src/factory.js

```diff
@@ -124,21 +133,25 @@ export default function factory(React, empty) {
           } else if(isAuthenticating) {
             return <LoadingComponent {...otherProps} />
           } else {
-            // Don't need to display anything because the user will be redirected
-            return React.createElement(empty)
+            // Unauthorized: Display FailureComponent or nothing
+            // if FailureComponent is undefined the user will never see
+            // this because they will be redirected to failureRedirectPath
+            return FailureComponent ? <FailureComponent authData={authData} {...otherProps} /> : React.createElement(empty)
```

Comment:
Hmmm.. I tried to avoid extra conditions in there but in that case I should add null explicitly. You're probably right, having an empty `<div>` when we should have nothing is probably wrong. It can break the layout in some cased and leak information in others.

Not sure about React Router's behavior in that case. Never being there.